packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -4133,12 +4135,22 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     _textInputConnection!.setComposingRect(composingRect);
   }
 
+  // Sends the current caret rect to the embedder's text input plugin.
+  //
+  // The position of the caret rect is updated on each change such that if the
+  // user initiates composing input, the current cursor rect can be used for
+  // the first character until the composing rect can be sent.
+  //
+  // On selection changes, the start of the selection is used. This ensures
+  // that regardless of the direction the selection was created, the cursor is
+  // set to the position where next text input occurs.
```

Comment:
nit: I think it will be helpful to future readers to give more context wrt why this has to do with "the next input". Maybe also add to the comment that the rects will be used to position the IME candidate view?

Reply:
also may be worth explaining why we are sending rects even when there's no caret (when the selection is not "collapsed").

Reply:
Good point. Done!